From c62a87e774a9aa8eb32be811e822481825048f23 Mon Sep 17 00:00:00 2001
From: elupus <elupus@xbmc.org>
Date: Mon, 2 Apr 2012 20:55:51 +0200
Subject: [PATCH 138/145] [dvdplayer] previous picture is invalidated by a
 call to codec reset (cherry picked from commit
 c0a367a4ecec766dc04e15d58f22c4860a04acca)

---
 xbmc/cores/dvdplayer/DVDPlayerVideo.cpp |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/cores/dvdplayer/DVDPlayerVideo.cpp b/xbmc/cores/dvdplayer/DVDPlayerVideo.cpp
index e28faaf..eb51038 100644
--- a/xbmc/cores/dvdplayer/DVDPlayerVideo.cpp
+++ b/xbmc/cores/dvdplayer/DVDPlayerVideo.cpp
@@ -413,6 +413,7 @@ void CDVDPlayerVideo::Process()
     {
       if(m_pVideoCodec)
         m_pVideoCodec->Reset();
+      picture.iFlags &= ~DVP_FLAG_ALLOCATED;
       m_packets.clear();
       m_started = false;
     }
@@ -420,6 +421,7 @@ void CDVDPlayerVideo::Process()
     {
       if(m_pVideoCodec)
         m_pVideoCodec->Reset();
+      picture.iFlags &= ~DVP_FLAG_ALLOCATED;
       m_packets.clear();
 
       m_pullupCorrection.Flush();
-- 
1.7.10

